### PR TITLE
#387 limit user's ability on experiment number of trials

### PR DESCRIPTION
**Pull request by Muhaddatha**
- Show dialog when user clicks 'save' or 'create' while creating or editing an experiment if the number of trials exceeds 100.
image

### CODE REVIEW

File: webapp/components/experiments/ExperimentEdit.js [hunk 1]
```diff
@@ -25,9 +25,12 @@ import Utils from '../../Utils';
 import ParameterMenu from './ParameterMenu';
 import useStyles from './ExperimentEditStyle';
 import * as ExperimentHelper from './ExperimentHelper';
+import DialogBox from '../general/DialogBox';
 
 const RANGE_VALUE = 0;
 const SUPPORTED_TYPES = [REAL_TYPE.INT, REAL_TYPE.FLOAT, REAL_TYPE.STR, REAL_TYPE.BOOL];
+const MAX_TRIALS = 100;
+let numberOfTrials = 1;
```
Comment (filippomc) [inline webapp/components/experiments/ExperimentEdit.js:33]: Why is numberOfTrials declared here?

File: webapp/components/experiments/ExperimentEdit.js [hunk 2]
```diff
@@ -512,43 +529,53 @@ const ExperimentEdit = (props) => {
...
           </Box>
         </Box>
-      </Box>
-    </GridLayout>
+      </GridLayout>
+      <DialogBox
+        open={trialNumberErrorDialogOpen}
+        onDialogResponse={() => setTrialNumberErrorDialogOpen(false)}
+        textForDialog={{
+          heading: 'Error - Number of trials is too large',
```
Comment (filippomc) [inline webapp/components/experiments/ExperimentEdit.js:573]: Were not trials renamed to "combinations"? cc @ddelpiano

**Review by filippomc — CHANGES_REQUESTED** [review]
I think we need to fix the naming as trials should not be used anymore

File: webapp/components/experiments/ExperimentEdit.js (hunk 2 above)
Comment (Muhaddatha) [reply webapp/components/experiments/ExperimentEdit.js:574]: It also says 'trials' when viewing an experiment's details. I can fix the wording here too once we know what word we're using.

image

Comment (filippomc) [reply webapp/components/experiments/ExperimentEdit.js:574]: I'm afraid there is some misalignment here, let's see what @ddelpiano and @enicolasgomez remember about this

Comment (ddelpiano) [reply webapp/components/experiments/ExperimentEdit.js:574]: it's a leftover from [this](https://app.zenhub.com/workspaces/netpyne---metacell-board-5bd615394b5806bc2bef3799/issues/metacell/netpyne-ui/368), my bad I missed @enicolasgomez comments down in the card, I will look at that and this in the afternoon today.

**Comment by ddelpiano** [comment]
Closing since I merged this inside feature/meta_migration

**Pull request closed without merging**